sdk/resourcemanager/compute/armcompute/utils_test.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/stretchr/testify/require"
+	"hash/fnv"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+)
+
+func authenticateTest(t *testing.T) (azcore.TokenCredential, *arm.ClientOptions) {
+	p := NewRecordingPolicy(t, &recording.RecordingOptions{UseHTTPS: true})
+	client, err := recording.GetHTTPClient(t)
+	require.NoError(t, err)
+
+	options := &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			PerCallPolicies: []policy.Policy{p},
+			Transport:       client,
+		},
+	}
+
+	var cred azcore.TokenCredential
+	if recording.GetRecordMode() != "playback" {
+		tenantId := lookupEnvVar("AZURE_TENANT_ID")
+		clientId := lookupEnvVar("AZURE_CLIENT_ID")
+		clientSecret := lookupEnvVar("AZURE_CLIENT_SECRET")
+		cred, err = azidentity.NewClientSecretCredential(tenantId, clientId, clientSecret, nil)
+		require.NoError(t, err)
+	} else {
+		cred = &fakeCredential{}
+	}
+
+	return cred, options
+}
+
+func lookupEnvVar(s string) string {
+	ret, ok := os.LookupEnv(s)
+	if !ok {
+		panic(fmt.Sprintf("Could not find env var: '%s'", s))
+	}
+	return ret
+}
+
+type recordingPolicy struct {
+	options recording.RecordingOptions
+	t       *testing.T
+}
+
+func (r recordingPolicy) Host() string {
+	if r.options.UseHTTPS {
+		return "localhost:5001"
+	}
+	return "localhost:5000"
+}
+
+func (r recordingPolicy) Scheme() string {
+	if r.options.UseHTTPS {
+		return "https"
+	}
+	return "http"
+}
+
+func NewRecordingPolicy(t *testing.T, o *recording.RecordingOptions) policy.Policy {
+	if o == nil {
+		o = &recording.RecordingOptions{UseHTTPS: true}
+	}
+	p := &recordingPolicy{options: *o, t: t}
+	return p
+}
+
+func (r *recordingPolicy) Do(req *policy.Request) (resp *http.Response, err error) {
+	if recording.GetRecordMode() != "live" && !recording.IsLiveOnly(r.t) {
+		originalURLHost := req.Raw().URL.Host
+		req.Raw().URL.Scheme = r.Scheme()
+		req.Raw().URL.Host = r.Host()
+		req.Raw().Host = r.Host()
+
+		req.Raw().Header.Set(recording.UpstreamURIHeader, fmt.Sprintf("%v://%v", r.Scheme(), originalURLHost))
+		req.Raw().Header.Set(recording.ModeHeader, recording.GetRecordMode())
+		req.Raw().Header.Set(recording.IDHeader, recording.GetRecordingId(r.t))
+	}
+	return req.Next()
+}
+
+func createRandomName(t *testing.T, prefix string) (string, error) {
+	h := fnv.New32a()
+	_, err := h.Write([]byte(t.Name()))
+	return prefix + fmt.Sprint(h.Sum32()), err
+}
+
+type fakeCredential struct {
+}
+
+func (f *fakeCredential) Do(req *policy.Request) (*http.Response, error) {
+	return req.Next()
+}
+
+func (f *fakeCredential) GetToken(ctx context.Context, options policy.TokenRequestOptions) (*azcore.AccessToken, error) {
+	return &azcore.AccessToken{
+		Token:     "faketoken",
+		ExpiresOn: time.Date(2040, time.January, 1, 1, 1, 1, 1, time.UTC),
+	}, nil
+}
+
+func createResourceGroup(t *testing.T,cred azcore.TokenCredential,option *arm.ClientOptions,subscriptionID,name,location string) (*armresources.ResourceGroup,func()) {
```

Comment:
My guess for why this is failing is that you don't actually need to run this when you're in playback, instead returning valid values for the playback test. You can add a snippet to the top of this function
``` 
    if recording.GetRecordMode() == recording.PlaybackMode {
        return &rg.ResourceGroup{/*fill in here */}, func() {}
    }
```